### ISSUE TITLE
Regression: Include list of pillar files under different keys in 2019.2

### ISSUE THREAD

**Issue by sploenix** [saltstack/salt#52134]
### Description of Issue/Question

When including different pillar files under different keys, the content of each file should be available under the specific key. After upgrading from 2018.3 to 2019.2 the content of the included files gets included under the last defined key. 
This behaviour affects the python3 as well as the python2 version of salt 2019.2.

### Setup
pillar/temp/a.sls
```
a:
  - 'Entry A'
```
pillar/temp/b.sls
```
b:
  - 'Entry B'
```
pillar/temp/init.sls

```
include:
  - temp.a:
      key: element:a
  - temp.b:
      key: element:b
```

### Steps to Reproduce Issue
```
salt-call pillar.get element
```

```
> local:
>     ----------
>     b:
>         ----------
>         a:
>             - Entry A
>         b:
>             - Entry B
```

expected behaviour:

```
> local:
>     ----------
>     a:
>         ----------
>         a:
>             - Entry A
>     b:
>         ----------
>         b:
>             - Entry B
```

### Versions Report
local:
    Salt Version:
               Salt: 2019.2.0
     
    Dependency Versions:
               cffi: Not Installed
           cherrypy: Not Installed
           dateutil: Not Installed
          docker-py: Not Installed
              gitdb: Not Installed
          gitpython: Not Installed
              ioflo: Not Installed
             Jinja2: 2.8
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: Not Installed
               Mako: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.5.6
       mysql-python: Not Installed
          pycparser: Not Installed
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 3.4.9 (default, Feb  5 2019, 14:36:09)
       python-gnupg: Not Installed
             PyYAML: 3.11
              PyZMQ: 15.3.0
               RAET: Not Installed
              smmap: Not Installed
            timelib: Not Installed
            Tornado: 4.4.2
                ZMQ: 4.1.4
     
    System Versions:
               dist: centos 7.6.1810 Core
             locale: UTF-8
            machine: x86_64
            release: 3.10.0-957.5.1.el7.x86_64
             system: Linux
            version: CentOS Linux 7.6.1810 Core

**Comment by commonism** [comment]
I guess it was broken in https://github.com/saltstack/salt/commit/fbab73a35c2a36e54ace814f2b45119e320b61e3

**Comment by dmurphy18** [comment]
@dwoz you will want to touch base with  @waynew as he recently made changes in this area with wildcard handling since 2019.2.0, and not closed similar duplicate #52303 for test cases when unit-testing fix.

**Comment by bertulus** [comment]
I just hit this issue in salt version 18.3.4 . Glad it is fixed already. Do you now when and if it will be available as an update ?  Thanks for the great work.

**Comment by Ch3LL** [comment]
@bertulus it will be included in the upcoming 2019.2.1 release. 

i'm surprised you are seeing this in the 2018.3.4 release as the commit that broke this was not in that release but only 2019.2.0. Is your use case possibly different? If so can you open a new issue for that release.

@dwoz is this ready to be closed?

**Comment by bertulus** [comment]
@Ch3LL It seems I was just stupid...  Sorry for your time and thanks for the reply.
